File: src/Entity/Trajetoffre.php
<?php

namespace App\Entity;

use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use App\Repository\TrajetoffreRepository;

#[ORM\Entity(repositoryClass: TrajetoffreRepository::class)]
class Trajetoffre
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $idtrajetoffre = null;
    

    
#[ORM\Column]

private ?int $limitekmoffre = null;
    

    #[ORM\Column(length: 255)]
    private ?string $addarriveoffre= null;


    #[ORM\Column(length: 255)]
    private ?string $adddepartoffre= null;
    

    #[ORM\Column]

private ?int $nbreescaleoffre = null;

    
#[ORM\Column]

private ?int $nbreoffre = null;
    
#[ORM\Column]
private ?string  $description = null;
    

    public function getIdtrajetoffre(): ?string
    {
        return $this->idtrajetoffre;
    }

    public function getLimitekmoffre(): ?int
    {
        return $this->limitekmoffre;
    }

    public function setLimitekmoffre(int $limitekmoffre): self
    {
        $this->limitekmoffre = $limitekmoffre;

        return $this;
    }

    public function getAddarriveoffre(): ?string
    {
        return $this->addarriveoffre;
    }

    public function setAddarriveoffre(?string $addarriveoffre): self
    {
        $this->addarriveoffre = $addarriveoffre;

        return $this;
    }

    public function getAdddepartoffre(): ?string
    {
        return $this->adddepartoffre;
    }

    public function setAdddepartoffre(?string $adddepartoffre): self
    {
        $this->adddepartoffre = $adddepartoffre;

        return $this;
    }

    public function getNbreescaleoffre(): ?int
    {
        return $this->nbreescaleoffre;
    }

    public function setNbreescaleoffre(?int $nbreescaleoffre): self
    {
        $this->nbreescaleoffre = $nbreescaleoffre;

        return $this;
    }

    public function getNbreoffre(): ?int
    {
        return $this->nbreoffre;
    }

    public function setNbreoffre(int $nbreoffre): self
    {
        $this->nbreoffre = $nbreoffre;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(): self
    {
        $this->description = $this->getAdddepartoffre() . ' - ' . $this->getAddarriveoffre();

        return $this;
    }

    public function __toString()
    {
        return $this->getDescription(); // Return the title of the task as the string representation
    }
}

File: src/com/mycompany/myapp/entities/TrajetOffre.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.mycompany.myapp.entities;

import java.util.ArrayList;
import java.util.List;

/**
 *
 * @author utilisateur
 */
public class TrajetOffre {
     private  long IdTrajetOffre ;

    private int LimiteKmOffre, NbreEscaleOffre, nbreOffre;
    private List<Offre> idoffres = new ArrayList<>();

    private String AddArriveOffre, AddDepartOffre, description;

    public TrajetOffre(int LimiteKmOffre, int NbreEscaleOffre, String AddArriveOffre, String AddDepartOffre) {
        this.LimiteKmOffre = LimiteKmOffre;
        this.NbreEscaleOffre = NbreEscaleOffre;
        this.AddArriveOffre = AddArriveOffre;
        this.AddDepartOffre = AddDepartOffre;
        this.description = description;
    }

    @Override
    public String toString() {
        return "TrajetOffre{" + "IdTrajetOffre=" + IdTrajetOffre + ", LimiteKmOffre=" + LimiteKmOffre + ", NbreEscaleOffre=" + NbreEscaleOffre + ", nbreOffre=" + nbreOffre + ", idoffres=" + idoffres + ", AddArriveOffre=" + AddArriveOffre + ", AddDepartOffre=" + AddDepartOffre + ", description=" + description + '}';
    }

    public TrajetOffre(long IdTrajetOffre, int LimiteKmOffre, int NbreEscaleOffre, String AddArriveOffre, String AddDepartOffre) {
        this.IdTrajetOffre = IdTrajetOffre;
        this.LimiteKmOffre = LimiteKmOffre;
        this.NbreEscaleOffre = NbreEscaleOffre;
        this.AddArriveOffre = AddArriveOffre;
        this.AddDepartOffre = AddDepartOffre;
    }

    public TrajetOffre() {
    }

    public void setIdTrajetOffre(long IdTrajetOffre) {
        this.IdTrajetOffre = IdTrajetOffre;
    }

    public void setLimiteKmOffre(int LimiteKmOffre) {
        this.LimiteKmOffre = LimiteKmOffre;
    }

    public void setNbreEscaleOffre(int NbreEscaleOffre) {
        this.NbreEscaleOffre = NbreEscaleOffre;
    }

    public void setNbreOffre(int nbreOffre) {
        this.nbreOffre = nbreOffre;
    }

    public void setIdoffres(List<Offre> idoffres) {
        this.idoffres = idoffres;
    }

    public void setAddArriveOffre(String AddArriveOffre) {
        this.AddArriveOffre = AddArriveOffre;
    }

    public void setAddDepartOffre(String AddDepartOffre) {
        this.AddDepartOffre = AddDepartOffre;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public long getIdTrajetOffre() {
        return IdTrajetOffre;
    }

    public int getLimiteKmOffre() {
        return LimiteKmOffre;
    }

    public int getNbreEscaleOffre() {
        return NbreEscaleOffre;
    }

    public int getNbreOffre() {
        return nbreOffre;
    }

    public List<Offre> getIdoffres() {
        return idoffres;
    }

    public String getAddArriveOffre() {
        return AddArriveOffre;
    }

    public String getAddDepartOffre() {
        return AddDepartOffre;
    }

    public String getDescription() {
        return description;
    }

}
